tests/manage/test_dynamic_pvc_accessmodes_with_reclaim_policies.py

```diff
@@ -0,0 +1,375 @@
+            secret_name = self.rbd_secret_obj.name
+            interface_name = self.cbp_obj.name
+
+        elif interface_type == constants.CEPHFILESYSTEM:
+            secret_name = self.cephfs_secret_obj.name
+            interface_name = helpers.get_cephfs_data_pool_name()
+
+        sc_obj = helpers.create_storage_class(
+            interface_type=interface_type,
+            interface_name=interface_name,
+            secret_name=secret_name,
+            reclaim_policy=reclaim_policy
+        )
+
+        # Create PVC with 'accessModes' 'ReadWriteOnce'
+        logger.info('Creating PVC with accessModes: ReadWriteOnce')
+        pvc_data = templating.load_yaml_to_dict(constants.CSI_PVC_YAML)
+        pvc_data['metadata']['name'] = helpers.create_unique_resource_name(
+            'test', 'pvc'
+        )
+        pvc_data['metadata']['namespace'] = self.namespace
+        pvc_data['spec']['storageClassName'] = sc_obj.name
+        pvc_data['spec']['accessModes'] = ['ReadWriteOnce']
+        pvc_obj = PVC(**pvc_data)
+        pvc_obj.create()
+
+        # Create two pods using same PVC
+        logger.info(f"Creating two pods using same PVC {pvc_obj.name}")
+
+        # Create first pod
+        if interface_type == constants.CEPHBLOCKPOOL:
+            pod_data = templating.load_yaml_to_dict(constants.CSI_RBD_POD_YAML)
+        elif interface_type == constants.CEPHFILESYSTEM:
+            pod_data = templating.load_yaml_to_dict(
+                constants.CSI_CEPHFS_POD_YAML
+            )
+
+        pod_data['metadata']['name'] = helpers.create_unique_resource_name(
+            'test', 'pod'
+        )
+        pod_data['metadata']['namespace'] = self.namespace
+        pod_data['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = (
+            pvc_obj.name
+        )
+        pod_obj = helpers.create_pod(**pod_data)
+
+        node_pod1 = pod_obj.get()['spec']['nodeName']
+
+        # Create second pod
+        # Try creating pod until it is on a different node than first pod
+        for retry in range(1, 6):
+            pod_data['metadata']['name'] = helpers.create_unique_resource_name(
+                'test', 'pod'
+            )
+            pod_obj2 = helpers.create_pod(wait=False, **pod_data)
+            node_pod2 = pod_obj2.get()['spec']['nodeName']
+            if node_pod1 != node_pod2:
+                break
+            logger.info(
+                f"Both pods are on same node. Deleting second pod and "
+                f"creating another pod. Retry count:{retry}"
+            )
+            pod_obj2.delete()
+            if retry == 5:
+                raise UnexpectedBehaviour(
+                    "Second pod is always created on same node as of first "
+                    "pod even after trying 5 times."
+                )
+
+        # Run IO on first pod
+        logger.info(f"Running IO on first pod {pod_obj.name}")
+        pod_obj.run_io('fs', '512M')
```

Comment:
the default run time is 'runtime=60'.
I think 30 is enough

Reply:
Done